server/mlabns/tests/test_resolver.py

```diff
@@ -78,6 +89,8 @@ def testGetCandidatesYesMemcache(self):
         self.testbed = testbed.Testbed()
         self.testbed.activate()
         self.testbed.init_memcache_stub()
+        _apply_namespace_bug_workaround()
```

Comment:
So there are multiple tests in this function? and the bug_workaround was called in between?

Is it possible to split this function into specific smaller ones?